dotnet/WPF/VisualLayerIntegration/VisualLayerIntegration/BarGraphUtility/BarBrushHelper.cs

```diff
@@ -30,111 +30,90 @@
-                var stop = compositor.CreateColorGradientStop(offset, color);
+                var offset = i / ((float)colors.Length - 1);
+                var stop = _compositor.CreateColorGradientStop(offset, color);
 
                 i++;
-                lgb.ColorStops.Add(stop);
+                linearGradientBrush.ColorStops.Add(stop);
             }
 
-            var brushes = new CompositionBrush[numBrushes];
-            for (int j = 0; j < numBrushes; j++)
-            {
-                brushes[j] = lgb;
-            }
-            return brushes;
+            return linearGradientBrush;
         }
 
-        internal CompositionBrush[] GenerateAmbientAnimatingPerBarLinearGradient(int numBrushes, List<Color> colors)
+        internal CompositionBrush GenerateAmbientAnimatingLinearGradient(Color[] colors)
         {
-            var lgb = compositor.CreateLinearGradientBrush();
-            lgb.RotationAngleInDegrees = 45;
+            var linearGradientBrush = _compositor.CreateLinearGradientBrush();
+            linearGradientBrush.RotationAngleInDegrees = 45;
 
-            int i = 0;
-            var animationDuration = 100;
+            var i = 0;
+            var animationDuration = TimeSpan.FromSeconds(100);
             foreach (Color color in colors)
             {
-                float offset = i / ((float)colors.Count - 1);
+                var offset = i / ((float)colors.Length - 1);
 
-                var stop = compositor.CreateColorGradientStop(offset, color);
-                lgb.ColorStops.Add(stop);
-                InitLGBAnimation(stop, animationDuration, 1.0f);
+                var stop = _compositor.CreateColorGradientStop(offset, color);
+                linearGradientBrush.ColorStops.Add(stop);
+                InitLinearGradientAnimation(stop, animationDuration, 1.0f);
 
                 // Create a second mirrored stop for all colors but the first.
                 if (offset > 0)
                 {
-                    var stop2 = compositor.CreateColorGradientStop(-offset, color);
-                    lgb.ColorStops.Add(stop2);
-                    InitLGBAnimation(stop2, animationDuration, 1.0f);
+                    var stop2 = _compositor.CreateColorGradientStop(-offset, color);
+                    linearGradientBrush.ColorStops.Add(stop2);
+                    InitLinearGradientAnimation(stop2, animationDuration, 1.0f);
                 }
 
                 i++;
             }
 
-            var brushes = new CompositionBrush[numBrushes];
-            for (int j = 0; j < numBrushes; j++)
-            {
-                brushes[j] = lgb;
-            }
-
-            return brushes;
+            return linearGradientBrush;
         }
 
-        private void InitLGBAnimation(CompositionColorGradientStop stop, int duration, float offsetAdjustment)
+        private void InitLinearGradientAnimation(CompositionColorGradientStop stop, TimeSpan duration, float offsetAdjustment)
```

Comment:
"Init" doesn't seem right - it's not the initial one and it doesn't initialize something. Maybe AnimateLinearGradientStopOffset(...).